### ISSUE TITLE
Add support of `wait` mode in sync commands

### ISSUE THREAD

**Issue by ta2-1** [translate/pootle#3727]
All sync tasks should be moved to RQ according to #3577. But some of our users can have problems if we apply it.

What if `update_stores` tasks shouldn't run for a `Project` before all `sync_stores` tasks are finished for this `Project`? How to handle it with sync commands and multiple workers?

We can create a command that checks if any sync job is running for a project (or a translation project). And then we can ask our users to adjust the way how they work with translation data.

Or we can add checking possibility to `update_stores` and `sync_stores` commands themselves.
For instance `./manage.py sync_stores --project=foo --wait` finishes after all sync tasks created for the `foo` project are finished.

**Comment by iafan** [comment]
I think that what we have discuss was:

1) to add a `--nowait` parameter for async operation. By default, the commands would run in a synchronous mode. This is to make the command-line API backward-compatiple.
2) in addition to `--nowait`, add the support for a separate command (let's call it `can_update_stores`) that would check if the project has no pending `sync_stores` jobs and is open for updates.

**Comment by dwaynebailey** [comment]
:+1: @iafan 

Though I would suggest maybe these alterations:

1. If you don't supply ``--nowait`` that we report that other jobs are busy and exit with a non-zero exit status.
2. That might make the need for ``can_update_stores`` if we can simply test for the exit status to determine if we exited because of running jobs.

**Comment by iafan** [comment]
Sometimes we would like to know whether to deal with the project within current sync cycle before we start updating stores, so I think having a standalone `can_update_stores` command makes sense anyway.

**Comment by ta2-1** [comment]
Since sync tasks are executed in a separate process created by rqworker it's impossible to wait when they finish via standard multi process API. Redis DB is a shared resource here. So we can check if a task is finished getting data from Redis keys directly.
Let’s call `TranslationProject` busy if there exists `STARTED`, `QUEUED` or `DEFERRED` sync job for it.
We can check if `TranslationProject` is busy easily. It’s necessary to check a status of a last job for this translation project.
In this issue we won’t support when `sync_stores` or `update_stores` commands are run from different places at the same moment. If we get result for `TranslationProject` that it’s not busy we don't take into account that somebody can run `sync_stores` or `update_stores` command for this `TranslationProject` and make it busy at the next moment after we get the result.

Another thing: if we run sync tasks for many `TranslationProject`s in one command (with `--project` option only or without `--language`, `--project` options), it can be not efficient to check all items in Redis DB very often.
Then we can put all sync tasks into a separate queue and check first task in the queue. If the task hasn't changed we have to wait more and can omit checking of all `TranslationProject`s.

**Comment by jleclanche** [comment]
No objections to landing this

**Comment by dwaynebailey** [comment]
@ta2-1 No objection.  Would it work to have a STARTING state to capture that small moment of two command starting together.  Some minor comments made

**Comment by ta2-1** [comment]
@dwaynebailey if you start two commands for the same set of `TranslationProject`s then it will wait till all jobs created within these two commands are finished.
I realized that we can create a separate queue in Redis for every executed command, avoid using of `while True:...` to wait and use [Redis blocking pop](http://redis.io/commands/blpop) from that queue instead.  In this way we are able to check if jobs which created within the definite command only are finished. So it can be done later.

**Comment by phlax** [comment]
i believe this is relevant to the old refresh_stats mechanism - please reopen @ta2-1 if its still relevant
